lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -60,18 +60,29 @@
       - Corresponds to the C(COMMAND) parameter of C(docker service create).
     type: raw
     version_added: 2.8
+  placement:
+    description:
+      - Configures service placement preferences and constraints.
+    suboptions:
+      constraints:
+        description:
+          - List of the service constraints.
+          - Corresponds to the C(--constraint) option of C(docker service create).
+        type: list
+      preferences:
+        description:
+          - List of the placement preferences as key value pairs.
+          - Corresponds to the C(--placement-pref) option of C(docker service create).
+          - Requires API version >= 1.27.
+        type: list
+    type: dict
+    version_added: "2.8"
   constraints:
     description:
       - List of the service constraints.
       - Corresponds to the C(--constraint) option of C(docker service create).
+      - Deprecated in 2.8, will be removed in 2.12. Use parameter C(placement) instead.
```

Comment:
I think it's better to name the full new name:
```suggestion
      - Deprecated in 2.8, will be removed in 2.12. Use parameter C(placement.constraints) instead.
```
Or something more explicit which tells the user to add `placements:`, and inside that `constraints:`.